staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -145,13 +149,26 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (r
 	return newObj, nil
 }
 
+// New sets an empty managedFields
+func (f *FieldManager) New(obj runtime.Object) (runtime.Object, error) {
+	// If the object doesn't have metadata, apply isn't allowed.
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get accessor: %v", err)
+	}
+	accessor.SetManagedFields([]metav1.ManagedFieldsEntry{})
+	return obj, nil
+}
+
 // Apply is used when server-side apply is called, as it merges the
 // object and update the managed fields.
 func (f *FieldManager) Apply(liveObj runtime.Object, patch []byte, fieldManager string, force bool) (runtime.Object, error) {
 	// If the object doesn't have metadata, apply isn't allowed.
-	if _, err := meta.Accessor(liveObj); err != nil {
+	accessor, err := meta.Accessor(liveObj)
```

Comment:
I don't understand why we have to do that, and the thing with the `New` function?

Reply:
Me too...